packages/jellyfish-api-core/__tests__/category/loan/updateVault.test.ts

```diff
@@ -0,0 +1,453 @@
+    vaultId = await alice.rpc.loan.createVault({
+      ownerAddress: address,
+      loanSchemeId: 'default'
+    })
+    await alice.generate(1)
+  })
+
+  afterAll(async () => {
+    await tGroup.stop()
+  })
+
+  beforeEach(async () => {
+    // Always update the vault back to default scheme
+    await alice.rpc.loan.updateVault(vaultId, {
+      ownerAddress: address,
+      loanSchemeId: 'default'
+    })
+    await alice.generate(1)
+
+    const data = await alice.rpc.loan.getVault(vaultId)
+    expect(data.loanSchemeId).toStrictEqual('default')
+  })
+
+  it('should updateVault', async () => {
+    const ownerAddress = await alice.generateAddress()
+    const updateVaultId = await alice.rpc.loan.updateVault(vaultId, {
+      ownerAddress,
+      loanSchemeId: 'scheme'
+    })
+    expect(typeof updateVaultId).toStrictEqual('string')
+    expect(updateVaultId.length).toStrictEqual(64)
+    await alice.generate(1)
+
+    const data = await alice.rpc.call('getvault', [vaultId], 'bignumber')
+    expect(data).toStrictEqual({
+      vaultId: vaultId,
+      loanSchemeId: 'scheme',
+      ownerAddress,
+      state: 'active',
+      collateralAmounts: [],
+      loanAmounts: [],
+      interestAmounts: [],
+      collateralValue: expect.any(BigNumber),
+      loanValue: expect.any(BigNumber),
+      interestValue: new BigNumber(0),
+      collateralRatio: new BigNumber(-1),
+      informativeRatio: new BigNumber(-1)
+    })
+  })
+
+  it('should updateVault with owner address only', async () => {
+    const ownerAddress = await alice.generateAddress()
+    await alice.rpc.loan.updateVault(vaultId, {
+      ownerAddress
+    })
+    await alice.generate(1)
+
+    const data = await alice.rpc.loan.getVault(vaultId)
+    expect(data.ownerAddress).toStrictEqual(ownerAddress)
+  })
+
+  it('should updateVault with loanSchemeId only', async () => {
+    await alice.rpc.loan.updateVault(vaultId, {
+      loanSchemeId: 'scheme'
+    })
+    await alice.generate(1)
+
+    const data = await alice.rpc.loan.getVault(vaultId)
+    expect(data.loanSchemeId).toStrictEqual('scheme')
+  })
+
+  it('should updateVault if the vault state is mayliquidate', async () => {
+    // Create another vault
+    const vaultId = await alice.rpc.loan.createVault({
+      ownerAddress: address,
+      loanSchemeId: 'default'
+    })
+    await alice.generate(1)
+
+    // Deposit to vault
+    await alice.rpc.loan.depositToVault({
+      vaultId, from: address, amount: '100@DFI'
+    })
+    await alice.generate(1)
+
+    // Take loan
+    await alice.rpc.loan.takeLoan({
+      vaultId,
+      amounts: '50@TSLA'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '0.9@DFI', currency: 'USD' }] })
+    await alice.generate(1)
+
+    const data = await alice.rpc.loan.getVault(vaultId)
+    expect(data.state).toStrictEqual(VaultState.MAY_LIQUIDATE)
```

Comment:
hmm... i thought will get frozen first...

Reply:
The price is always valid as Price drops from 1 to 0.9 (Only 10% difference)